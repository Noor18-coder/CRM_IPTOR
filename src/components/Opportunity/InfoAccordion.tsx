import React from 'react'
import {Accordion, Card}  from 'react-bootstrap'

const InfoAccordion = () => {
    return (
        // <!-- INFO ACCORDION MAIN SECTION START -->
        
        <section className="sec-info-accordion">
        {/*    <!-- ACCORDION START --> */}
          <Accordion  defaultActiveKey="0">
          <Card>
            <Accordion.Toggle as={Card.Link} eventKey="1">
            Basic Info
            </Accordion.Toggle>
            <Accordion.Collapse eventKey="1">
              <Card.Body className="accr-body-container">  
                 <ul className="list-inline bdy-list-item">
                  <li className="list-inline-item"><span>Opp owner</span>Marek Kandybo</li>
                  <li className="list-inline-item"><span>Opp Type</span>License</li>
                  <li className="list-inline-item"><span>Opp Number</span>076123</li>
                  <li className="list-inline-item"><span>Region</span>north america</li>
                  <li className="list-inline-item"><span>RLF Year 1</span>-</li>
                  <li className="list-inline-item"><span>M&amp; Year 1</span>-</li>
                  <li className="list-inline-item"><span>PS Value</span>-</li>
                </ul></Card.Body>
            </Accordion.Collapse>
          </Card>
          </Accordion>

          <Accordion  defaultActiveKey="0">
          <Card>
            <Accordion.Toggle as={Card.Link} eventKey="1">
            Item Info
            </Accordion.Toggle>
            <Accordion.Collapse eventKey="1">
              <Card.Body className="accr-body-container">   
                <ul className="list-inline bdy-list-item">
                  <li className="list-inline-item"><span>Opp owner</span>Marek Kandybo</li>
                  <li className="list-inline-item"><span>Opp Type</span>License</li>
                  <li className="list-inline-item"><span>Opp Number</span>076123</li>
                  <li className="list-inline-item"><span>Region</span>north america</li>
                  <li className="list-inline-item"><span>RLF Year 1</span>-</li>
                  <li className="list-inline-item"><span>M&amp; Year 1</span>-</li>
                  <li className="list-inline-item"><span>PS Value</span>-</li>
                </ul>
              </Card.Body>
            </Accordion.Collapse>
          </Card>
          </Accordion>


          <Accordion  defaultActiveKey="0">
          <Card>
            <Accordion.Toggle as={Card.Link} eventKey="1">
            More Info
            </Accordion.Toggle>
            <Accordion.Collapse eventKey="1">
              <Card.Body className="accr-body-container">   
              <ul className="list-inline bdy-list-item">
                  <li className="list-inline-item"><span>Opp owner</span>Marek Kandybo</li>
                  <li className="list-inline-item"><span>Opp Type</span>License</li>
                  <li className="list-inline-item"><span>Opp Number</span>076123</li>
                  <li className="list-inline-item"><span>Region</span>north america</li>
                  <li className="list-inline-item"><span>RLF Year 1</span>-</li>
                  <li className="list-inline-item"><span>M&amp; Year 1</span>-</li>
                  <li className="list-inline-item"><span>PS Value</span>-</li>
                </ul>
              </Card.Body>
            </Accordion.Collapse>
          </Card>
          </Accordion>

          <Accordion  defaultActiveKey="0">
          <Card>
            <Accordion.Toggle as={Card.Link} eventKey="1">
            Contact Info
            </Accordion.Toggle>
            <Accordion.Collapse eventKey="1">
              <Card.Body className="accr-body-container">   
              <ul className="list-inline bdy-list-item">
                  <li className="list-inline-item"><span>Opp owner</span>Marek Kandybo</li>
                  <li className="list-inline-item"><span>Opp Type</span>License</li>
                  <li className="list-inline-item"><span>Opp Number</span>076123</li>
                  <li className="list-inline-item"><span>Region</span>north america</li>
                  <li className="list-inline-item"><span>RLF Year 1</span>-</li>
                  <li className="list-inline-item"><span>M&amp; Year 1</span>-</li>
                  <li className="list-inline-item"><span>PS Value</span>-</li>
                </ul>
              </Card.Body>
            </Accordion.Collapse>
          </Card>
          </Accordion>
          
         {/* <!-- ACCORDION END --> */}
        </section>
        // <!-- INFO ACCORDION MAIN SECTION END -->
    )
}

export default InfoAccordion
