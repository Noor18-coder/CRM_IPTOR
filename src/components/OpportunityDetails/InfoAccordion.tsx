import React from 'react';
import { includes } from 'lodash';
import { Accordion, Card } from 'react-bootstrap';
import { useSelector } from 'react-redux';
import { AppState } from '../../store/store';
import ImageConfig from '../../config/ImageConfig';
import { getDateInFormat } from '../../helpers/utilities/lib';

import {
  OpportunityDetailsBasicInfo,
  AttributeField,
  IAttributesList,
  AttributeValueObject,
  OpportunityEditOptions,
  OpportunityType,
} from '../../helpers/Api/models';

interface Props {
  title: string;
  data: OpportunityDetailsBasicInfo[];
  openEditOpportunity: () => void;
}

export const InfoAccordion: React.FC<Props> = ({ title, data, openEditOpportunity }) => {
  const state: OpportunityEditOptions = useSelector((appState: AppState) => appState.opportuntyDetails.editOportunity);

  const [activeClass, setActiveClass] = React.useState('');
  const toggleAccordion = () => {
    setActiveClass(activeClass === '' ? 'active' : '');
  };
  return (
    <Accordion defaultActiveKey="0">
      <Card>
        <Accordion.Toggle className={activeClass} onClick={toggleAccordion} as={Card.Link} eventKey="1">
          {title}
        </Accordion.Toggle>
        <Accordion.Collapse eventKey="1">
          <div className="accr-body-container">
            {state.allowEdit ? (
              <button type="button" className="edit-icon" onClick={openEditOpportunity}>
                <img src={ImageConfig.EDIT_ICON} alt="edit" />
              </button>
            ) : null}
            <ul className="list-inline bdy-list-item accr-list-columns">
              {data.map((obj: OpportunityDetailsBasicInfo) => {
                return (
                  <li className="list-inline-item">
                    <span>{obj.description}</span>
                    {obj.attributeValue ? `${obj.attributeValue}` : 'NA'}
                  </li>
                );
              })}
            </ul>
          </div>
        </Accordion.Collapse>
      </Card>
    </Accordion>
  );
};

interface GroupAccordianProps {
  title: string;
  openEditForm: (groupName: string) => void;
  oppType: string;
}

export const InfoAccordionGroups: React.FC<GroupAccordianProps> = ({ title, openEditForm, oppType }) => {
  const state: AppState = useSelector((appState: AppState) => appState);
  const { attributes, opportunityDefaultParams } = state.opportuntyDetails;
  const [moreInformationGroups, setMoreInformationGroups] = React.useState<IAttributesList[]>([]);
  const [activeClass, setActiveClass] = React.useState('');

  const toggleAccordion = () => {
    setActiveClass(activeClass === '' ? 'active' : '');
  };

  const setAllFields = (_opptyType: string) => {
    const groups = new Set(
      state.enviornmentConfigs.opportunityAttributes.map((obj: any) => {
        return obj.group;
      })
    );

    const recordType: OpportunityType | undefined = state.enviornmentConfigs.crmOpportunityTypes.find((obj: OpportunityType) => {
      return obj.oppRecordType === _opptyType;
    });

    const optionalFields = recordType && recordType.OPTIONAL_FIELDS ? recordType.OPTIONAL_FIELDS : [];
    const mandatoryFields = recordType && recordType.MANDATORY_FIELDS ? recordType.MANDATORY_FIELDS : [];
    const strArray = [...optionalFields, ...mandatoryFields];
    const tempAttributes: AttributeField[] = state.enviornmentConfigs.opportunityAttributes || [];

    if (strArray.length) {
      const response: IAttributesList[] = [];
      const filteredAttributes: AttributeField[] = tempAttributes.filter((obj: AttributeField) => {
        return includes(strArray, obj.attributeType);
      });

      groups.forEach((group: string) => {
        const items: AttributeField[] = filteredAttributes.filter((obj) => obj.group === group);
        if (items.length) {
          response.push({ group, items });
        }
      });

      setMoreInformationGroups(response);
    } else {
      const response: IAttributesList[] = [];
      groups.forEach((group: string) => {
        const groupAttributes: IAttributesList = { group, items: [] };
        groupAttributes.items = state.enviornmentConfigs.opportunityAttributes.filter((obj) => obj.group === group);
        response.push(groupAttributes);
      });
      setMoreInformationGroups(response);
    }
  };

  React.useEffect(() => {
    setAllFields(oppType);
  }, [oppType]);

  return (
    <Accordion defaultActiveKey="0">
      <Card>
        <Accordion.Toggle className={activeClass} onClick={toggleAccordion} as={Card.Link} eventKey="1">
          {title}
        </Accordion.Toggle>
        <Accordion.Collapse eventKey="1">
          <>
            {opportunityDefaultParams.oppRecordType && moreInformationGroups && moreInformationGroups?.length
              ? moreInformationGroups.map((key: IAttributesList) => {
                  return (
                    <DisplayGroup
                      title={key.group}
                      fields={key.items}
                      data={attributes}
                      openEditForm={openEditForm}
                      allowEdit={state.opportuntyDetails.editOportunity.allowEdit}
                    />
                  );
                })
              : null}
          </>
        </Accordion.Collapse>
      </Card>
    </Accordion>
  );
};

interface GroupData {
  title: string;
  fields: AttributeField[];
  data: AttributeValueObject[];
  openEditForm: (key: string) => void;
  allowEdit?: boolean;
}

export const DisplayGroup: React.FC<GroupData> = ({ title, fields, data, openEditForm, allowEdit }) => {
  const getValue = (obj: AttributeField) => {
    const dataObject = data.find((object: AttributeValueObject) => {
      return object.attributeType === obj.attributeType;
    });
    const field = fields.find((object: AttributeField) => {
      return object.attributeType === obj.attributeType;
    });
    const value = dataObject && dataObject.attributeValue ? dataObject.attributeValue : '';
    if (field?.valueFormatDesc === 'DATE') {
      if (value !== '') {
        const year = value.substring(0, 4);
        const month = value.substring(4, 6);
        const day = value.substring(6, 8);
        const tempDate = `${year}-${month}-${day}`;
        return getDateInFormat(new Date(tempDate));
      } else {
        return '--';
      }
    } else if (field?.valueFormatDesc === 'BOOLEAN') {
      return (
        <span className="checkbox-label">
          <label className="switch value-checkbox">
            <input type="checkbox" tabIndex={0} checked={value === 'Y'} />
            <span className="slider round disabled-checkbox">&nbsp;</span>
          </label>
        </span>
      );
    }
    return value;
  };

  return (
    <div className="more-info-group-container">
      <div className="more-info-group-name">
        {title}
        {allowEdit ? (
          <button type="button" className="edit-moreinfo" onClick={() => openEditForm(title)}>
            <img src={ImageConfig.EDIT_ICON} alt="edit" />
          </button>
        ) : null}
      </div>
      <div className="accr-body-container">
        <ul className="list-inline bdy-list-item accr-list-columns">
          {fields.map((obj: AttributeField) => {
            if (obj.uniqueRecord) {
              return (
                <li className="list-inline-item">
                  <span>{obj.description}</span>
                  {getValue(obj)}
                </li>
              );
            } else {
              const dataObj: AttributeValueObject[] = data.filter((dataObject: AttributeValueObject) => {
                return dataObject.attributeType === obj.attributeType;
              });
              if (dataObj.length) {
                return dataObj.map((elem: AttributeValueObject) => {
                  return (
                    <li className="list-inline-item">
                      <span>{obj.description}</span>
                      {elem.attributeValue}
                    </li>
                  );
                });
              } else {
                return (
                  <li className="list-inline-item">
                    <span>{obj.description}</span>
                    --
                  </li>
                );
              }
            }
          })}
        </ul>
      </div>
    </div>
  );
};

interface GroupMobileData {
  fields: AttributeField[];
  data: AttributeValueObject[];
}

export const DisplayGroupMobile: React.FC<GroupMobileData> = ({ fields, data }) => {
  const getValue = (obj: AttributeField) => {
    const dataObject = data.find((object: AttributeValueObject) => {
      return object.attributeType === obj.attributeType;
    });

    const value = dataObject && dataObject.attributeValue ? dataObject.attributeValue : '';

    if (!value) {
      return '-';
    }

    if (value && obj?.valueFormatDesc === 'DATE') {
      const year = value.substring(0, 4);
      const month = value.substring(4, 6);
      const day = value.substring(6, 8);
      return `${year}-${month}-${day}`;
    } else if (obj?.valueFormatDesc === 'BOOLEAN') {
      return value ? (value === 'Y' ? 'No' : 'Yes') : '-';
    }
    return value;
  };
  return (
    <div className="more-info-group-container">
      <div className="accr-body-container">
        <ul className="list-inline bdy-list-item">
          {fields.map((obj: AttributeField) => {
            return (
              <li className="list-inline-item">
                <span>{obj.description}</span>
                {getValue(obj)}
              </li>
            );
          })}
        </ul>
      </div>
    </div>
  );
};

export const AccordianForMobileWithGroups: React.FC<GroupAccordianProps> = ({ title, openEditForm, oppType }) => {
  const state: AppState = useSelector((appState: AppState) => appState);
  const { attributes } = state.opportuntyDetails;
  const [moreInformationGroups, setMoreInformationGroups] = React.useState<IAttributesList[]>([]);
  const [activeClass, setActiveClass] = React.useState('');
  const toggleAccordion = () => {
    setActiveClass(activeClass === '' ? 'active' : '');
  };

  // React.useEffect(() => {
  //   const groups = new Set(
  //     state.enviornmentConfigs.opportunityAttributes.map((obj: any) => {
  //       return obj.group;
  //     })
  //   );
  //   const response: IAttributesList[] = [];
  //   groups.forEach((group: string) => {
  //     const groupName = group.toLowerCase();
  //     const groupAttributes: IAttributesList = { group: groupName, items: [] };
  //     groupAttributes.items = state.enviornmentConfigs.opportunityAttributes.filter((obj) => obj.group.toLowerCase() === groupName);
  //     response.push(groupAttributes);
  //   });
  //   setMoreInformationGroups(response);
  // }, []);

  const setAllFields = (_opptyType: string) => {
    const groups = new Set(
      state.enviornmentConfigs.opportunityAttributes.map((obj: any) => {
        return obj.group;
      })
    );

    const recordType: OpportunityType | undefined = state.enviornmentConfigs.crmOpportunityTypes.find((obj: OpportunityType) => {
      return obj.oppRecordType === _opptyType;
    });

    const optionalFields = recordType && recordType.OPTIONAL_FIELDS ? recordType.OPTIONAL_FIELDS : [];
    const mandatoryFields = recordType && recordType.MANDATORY_FIELDS ? recordType.MANDATORY_FIELDS : [];
    const strArray = [...optionalFields, ...mandatoryFields];
    const tempAttributes: AttributeField[] = state.enviornmentConfigs.opportunityAttributes || [];

    if (strArray.length) {
      const response: IAttributesList[] = [];
      const filteredAttributes: AttributeField[] = tempAttributes.filter((obj: AttributeField) => {
        return includes(strArray, obj.attributeType);
      });

      groups.forEach((group: string) => {
        const items: AttributeField[] = filteredAttributes.filter((obj) => obj.group === group);
        if (items.length) {
          response.push({ group, items });
        }
      });

      setMoreInformationGroups(response);
    } else {
      const response: IAttributesList[] = [];
      groups.forEach((group: string) => {
        const groupAttributes: IAttributesList = { group, items: [] };
        groupAttributes.items = state.enviornmentConfigs.opportunityAttributes.filter((obj) => obj.group === group);
        response.push(groupAttributes);
      });
      setMoreInformationGroups(response);
    }
  };

  React.useEffect(() => {
    setAllFields(oppType);
  }, [oppType]);

  return (
    <div className="opp-moreinfo-sec">
      {title}
      <>
        {moreInformationGroups && moreInformationGroups.length
          ? moreInformationGroups.map((obj: IAttributesList) => {
              return (
                <Accordion defaultActiveKey="0">
                  <Card>
                    <Accordion.Toggle className={activeClass} onClick={toggleAccordion} as={Card.Link} eventKey="1">
                      {obj.group}
                      {state.opportuntyDetails.editOportunity.allowEdit ? (
                        <button type="button" className="mob-edit-icon" onClick={() => openEditForm(obj.group)}>
                          <img src={ImageConfig.EDIT_ICON} alt="edit" />
                        </button>
                      ) : null}
                    </Accordion.Toggle>
                    <Accordion.Collapse eventKey="1">
                      <DisplayGroupMobile fields={obj.items} data={attributes} />
                    </Accordion.Collapse>
                  </Card>
                </Accordion>
              );
            })
          : null}
      </>
    </div>
  );
};
