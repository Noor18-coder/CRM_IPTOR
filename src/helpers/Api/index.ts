export * from './ApiRequest';
export * from './User';
export * from './CompanyInfo';
export * from './CurrencyInfo';
export * from './StagesInfo';
export * from './OpportunityType';
export * from './Attributes';
