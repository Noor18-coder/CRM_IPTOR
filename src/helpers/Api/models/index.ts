export * from './UserItem';
export * from './CompanyInfo';
export * from './ApiRequestControls';
export * from './Opportunity';
export * from './OpportunityDetails';
export * from './Users';
export * from './StageInfo';
export * from './Contacts';
export * from './Product';
export * from './Customer';
export * from './OpportunityTypes';
export * from './CustomerDetails';
export * from './AddOpportunity';
export * from './Items';
export * from './OpportunityAttributes';
export * from './Currency';
export * from './Attributes';
export * from './AddCustomer';
export * from './Countries';
export * from './ForeCastInfo';
