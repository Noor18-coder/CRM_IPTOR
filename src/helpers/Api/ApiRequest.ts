/**
 * Api Request Object
 */
import { ApiRequestControls } from './models';

export class ApiRequest<T> {
  IptorAPI: string;

  id: string;

  constructor(public method: string, public params?: T, public control?: ApiRequestControls) {
    this.IptorAPI = process.env.REACT_APP_IPTOR_API_VERSION ?? '';
    this.id = process.env.REACT_APP_IPTOR_API_ID ?? '';
  }
}
