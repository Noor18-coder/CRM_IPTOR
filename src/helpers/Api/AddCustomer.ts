import axios, { AxiosResponse } from 'axios';
import { get } from 'lodash';
import { AddBusinessPartnerDefaultParams, AddBusinessPartnerResponse, SaveUserDefinedFieldParam } from './models';
import { ApiRequest } from './ApiRequest';

export default class AddCustomerApi {
  /** API Method */
  private static apiMethod = 'crmBusinessPartner.add';

  private static updateApiMethod = 'crmBusinessPartner.update';

  private static saveAttributeValueMethod = 'mopAttribute.add';

  private static saveAttributeValueMethodFile = 'SRONAM';

  /**
   * Helper function to fetch Business Partner Info
   * API Method: crmBusinessPartner.add
   * */
  static async add(businessPartner: AddBusinessPartnerDefaultParams): Promise<AddBusinessPartnerResponse> {
    const requestData = new ApiRequest<AddBusinessPartnerDefaultParams>(this.apiMethod, businessPartner);
    const response = await axios.post<AxiosResponse>('/api/service', requestData);
    return get(response, 'data');
  }

  static async update(businessPartner: AddBusinessPartnerDefaultParams): Promise<AddBusinessPartnerResponse> {
    const requestData = new ApiRequest<AddBusinessPartnerDefaultParams>(this.updateApiMethod, businessPartner);
    const response = await axios.post<AxiosResponse>('/api/service', requestData);
    return get(response, 'data');
  }

  static async addAttributes(businessPartnerId: string, attributeType: string, attributeValue: string | number): Promise<AddBusinessPartnerResponse> {
    const params: SaveUserDefinedFieldParam = {
      attributeType,
      parentFile: this.saveAttributeValueMethodFile,
      parentId: businessPartnerId,
      attributeValue,
    };
    const requestData = new ApiRequest<SaveUserDefinedFieldParam>(this.saveAttributeValueMethod, params);
    const response = await axios.post<AxiosResponse>('/api/service', requestData);
    return get(response, 'data');
  }
}
