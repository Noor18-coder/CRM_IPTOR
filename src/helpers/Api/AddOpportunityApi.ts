import axios, { AxiosResponse } from 'axios';
import { AddOpportunityDefaultParams, AddOpportunityResponse, SaveUserDefinedFieldParam, AddItemToOpportunityParams, AddItemToOpportunityResponse, AddCustomerContactParams, AddCustomerContactRequestParams } from './models';
import { ApiRequest } from './ApiRequest';
import { get } from 'lodash';

export default class AddOpportunityApi {
  /** API Method */
  private static apiMethod: string = 'mopOpportunity.add';
  private static saveAttributeValueMethod: string = 'mopAttribute.add';
  private static saveAttributeValueMethodFile: string = 'SROMOPH';
  private static saveItemMethod: string = 'mopItem.add';
  private static saveContactFileName: string = 'SROMOPCH';
  private static saveContactMethod: string = 'mopContact.add';

  /**
   * Helper function to fetch Business Partner Info
   * API Method: mopOpportunity.get
   * @param freeTextSearch search term
   * 
   */
  static async add(opportunity: AddOpportunityDefaultParams): Promise<AddOpportunityResponse> {
    const requestData = new ApiRequest<AddOpportunityDefaultParams>(this.apiMethod, opportunity);
    const response = await axios.post<AxiosResponse>('/api/service', requestData);
    return get(response, 'data');
  }

  /**
   * Helper function to fetch Business Partner Info
   * API Method: mopOpportunity.get
   * @param freeTextSearch search term
   * 
   */
  static async addAttributes(opportunityId: string, attributeType: string, attributeValue: string | number): Promise<AddOpportunityResponse> {
    const params: SaveUserDefinedFieldParam = {
      attributeType: attributeType,
      parentFile: this.saveAttributeValueMethodFile,
      parentId: opportunityId,
      attributeValue: attributeValue
    }
    const requestData = new ApiRequest<SaveUserDefinedFieldParam>(this.saveAttributeValueMethod, params);
    const response = await axios.post<AxiosResponse>('/api/service', requestData);
    return get(response, 'data');
  }

  /**
   * Helper function to fetch Business Partner Info
   * API Method: mopOpportunity.get
   * @param freeTextSearch search term
   * 
   */
  static async addItem(opportunityId: string, item: string, quantity: number, unit:string): Promise<AddItemToOpportunityResponse> {
    const params: AddItemToOpportunityParams = {
      parentFile: this.saveAttributeValueMethodFile,
      parentId: opportunityId,
      item: item,
      quantity: quantity,
      unit:unit
    }
    const requestData = new ApiRequest<AddItemToOpportunityParams>(this.saveItemMethod, params);
    const response = await axios.post<AddItemToOpportunityResponse>('/api/service', requestData);
    return get(response, 'data');
  }

  /**
   * Helper function to fetch Business Partner Info
   * API Method: mopOpportunity.get
   * @param freeTextSearch search term
   * 
   */
   static async addContacts(params:AddCustomerContactParams): Promise<AxiosResponse> {
    const request:AddCustomerContactRequestParams  = {
      contactParentFile:this.saveAttributeValueMethodFile,
      contactParentId:params.contactPArentId,
      contactPerson:params.contactPerson,
      phone:params.phone,
      mobile:params.mobile,
      contactDC:'',
      whatsApp:'',
      linkedin:'',
      fax:'',
      email:params.email
    }

    const requestData = new ApiRequest<AddCustomerContactRequestParams>(this.saveContactMethod, request);
    const response = await axios.post<AxiosResponse>('/api/service', requestData);
    console.log(response);
    return get(response, 'data');
  }
}
