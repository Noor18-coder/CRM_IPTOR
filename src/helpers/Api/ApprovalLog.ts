import axios, { AxiosResponse } from 'axios';
import { get } from 'lodash';
import { ApprovalLogAddRequestParams, ApprovalLogAddResponse, DeleteApprovalLogsParams } from './models';
import { ApiRequest } from './ApiRequest';

export class ApprovalLog {
  /** API Method */
  private static apiMethod = 'crmOpportunityApprovalLog.add';

  private static deleteApprovalLogs = 'crmOpportunityApprovalLog.delete';

  /**
   * Helper function to fetch sales order list
   * @param params user id
   * @returns information about user
   */
  static async addApprovalLog(params: ApprovalLogAddRequestParams): Promise<ApprovalLogAddResponse> {
    const requestData = new ApiRequest<ApprovalLogAddRequestParams>(this.apiMethod, params);
    const response = await axios.post<ApprovalLogAddResponse>('/api/service', requestData);
    return get<AxiosResponse<ApprovalLogAddResponse>, 'data'>(response, 'data');
  }

  static async deleteOpportunityApprovalLogs(params: DeleteApprovalLogsParams): Promise<ApprovalLogAddResponse> {
    const requestData = new ApiRequest<DeleteApprovalLogsParams>(this.deleteApprovalLogs, params);
    const response = await axios.post<ApprovalLogAddResponse>('/api/service', requestData);
    return get(response, 'data.data.items', {});
  }
}
