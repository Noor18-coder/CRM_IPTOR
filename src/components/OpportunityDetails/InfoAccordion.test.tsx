import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { InfoAccordion, InfoAccordionGroups, DisplayGroup, DisplayGroupMobile, AccordianForMobileWithGroups } from './InfoAccordion';
import StoreMock from '../../mocks/Store.mock';
import { OpportunityDetailsBasicInfo, AttributeField, AttributeValueObject } from '../../helpers/Api/models';

const store = StoreMock.createInitialStore();
const data = [] as OpportunityDetailsBasicInfo[];
const fields = [] as AttributeField[];
const groupData = [] as AttributeValueObject[];
const openEditOpportunity = jest.fn();
const openEditForm = jest.fn();

describe('[OpportunityDetails] InfoAccordion', () => {
  it('should renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <InfoAccordion title="title" data={data} openEditOpportunity={openEditOpportunity} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('[OpportunityDetails] InfoAccordionGroups', () => {
  it('should renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <InfoAccordionGroups title="title" oppType="oppType" openEditForm={openEditForm} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('[OpportunityDetails] DisplayGroup', () => {
  it('should renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <DisplayGroup title="title" fields={fields} data={groupData} openEditForm={openEditForm} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('[OpportunityDetails] DisplayGroupMobile', () => {
  it('should renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <DisplayGroupMobile fields={fields} data={groupData} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('[OpportunityDetails] AccordianForMobileWithGroups', () => {
  it('should renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <AccordianForMobileWithGroups title="title" oppType="oppType" openEditForm={openEditForm} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
