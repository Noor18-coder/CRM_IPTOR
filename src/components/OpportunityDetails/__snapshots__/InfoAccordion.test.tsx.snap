// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[OpportunityDetails] AccordianForMobileWithGroups should renders correctly 1`] = `
<div
  className="opp-moreinfo-sec"
>
  title
</div>
`;

exports[`[OpportunityDetails] DisplayGroup should renders correctly 1`] = `
<div
  className="more-info-group-container"
>
  <div
    className="more-info-group-name"
  >
    title
  </div>
  <div
    className="accr-body-container"
  >
    <ul
      className="list-inline bdy-list-item accr-list-columns"
    />
  </div>
</div>
`;

exports[`[OpportunityDetails] DisplayGroupMobile should renders correctly 1`] = `
<div
  className="more-info-group-container"
>
  <div
    className="accr-body-container"
  >
    <ul
      className="list-inline bdy-list-item"
    />
  </div>
</div>
`;

exports[`[OpportunityDetails] InfoAccordion should renders correctly 1`] = `
<div
  className="accordion"
>
  <div
    className="card"
  >
    <a
      className="card-link"
      onClick={[Function]}
    >
      title
    </a>
    <div
      aria-expanded={null}
      className="collapse"
    >
      <div
        className="accr-body-container"
      >
        <ul
          className="list-inline bdy-list-item accr-list-columns"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`[OpportunityDetails] InfoAccordionGroups should renders correctly 1`] = `
<div
  className="accordion"
>
  <div
    className="card"
  >
    <a
      className="card-link"
      onClick={[Function]}
    >
      title
    </a>
    <div
      aria-expanded={null}
      className="collapse"
    />
  </div>
</div>
`;
